R/calcIndustry_CCS_limits.R

```diff
@@ -0,0 +1,178 @@
+#' Calculate Limits on Industry CCS Capacities
+#'
+#' @md
+#' @details
+#' The limits on industry CCS capacities are calculated from data of the
+#' [Global Status of CCS 2023](zotero://select/items/3_E5GNNPZ8) report (through
+#' [`readGlobalCCSinstitute()`].  CCS projects are
+#' - filtered for valid (i.e. not "Under Evaluation") data for "Operation date"
+#'   and "CO~2~ capture capacity"
+#' - assigned to REMIND industry subsectors according to `facility_subsector`,
+#'   which defaults to
+#'   | Facility Industry               | subsector |
+#'   |:--------------------------------|:----------|
+#'   | Cement                          | cement    |
+#'   | Chemical                        | chemicals |
+#'   | Hydrogen / Ammonia / Fertiliser | chemicals |
+#'   | Ethan                           | chemicals |
+#'   | Iron and Steel Production       | steel     |
+#'
+#' - weighted by lifecycle stage according to `stage_weight`, which defaults to
+#'    | Lifecycle stage      | weight |
+#'    |:---------------------|-------:|
+#'    | Operational          | 100 %  |
+#'    | In construction      | 100 %  |
+#'    | Advanced development |  50 %  |
+#'    | Early development    |  20 %  |
+#'
+#' The resulting project capacities constitute the limits on industry subsector
+#' CCS capacity for 2030.  The limit on CCS capacities for regions (or countries
+#' if `region_mapping` is `NULL`) is set to a value of total 2030 subsector CCS
+#' capacity, times the regions share in 2030 subsector activity (e.g. cement
+#' production)
+#' - in 2035 if the region as some CCS capacity in 2030 in a different industry
+#'   subsector, or
+#' - in 2040 if the region has no industry CCS capacity in 2030 at all.
+#'
+#' CCS capacities are increase by the annual growth factor (default: 20 %/p.a.).
+#'
+#' @param annual_growth_factor Annual growth factor of CCS capacity limits,
+#'     defaults to 1.2 (20 % annually).
+#' @param stage_weight  A named vector of weight factors for different lifecycle
+#'     stages.  See Details.
+#' @param facility_subsector A named vector mapping the "Facility Industry" of
+#'     CCS projects to REMIND industry subsectors.  See Details.
+#' @param region_mapping A data frame with columns `iso3c` and `region` detailing
+#'     the regional resolution on which data should be extrapolated.  If `NULL`
+#'     (the default), extrapolation is done at the country level.
+#'
+#' @return A list with a [`magpie`][magclass::magclass] object `x`, `weight`,
+#'   `unit`, `description`, and `min`.
+#'
+#' @author Michaja Pehl
+#'
+#' @importFrom dplyr %>% group_by full_join inner_join left_join mutate pull
+#'     rename select summarise tibble ungroup
+#' @importFrom quitte add_countrycode_ madrat_mule magclass_to_tibble
+#' @importFrom readr read_delim
+#' @importFrom tidyr complete pivot_longer
+#'
+#' @export
+calcIndustry_CCS_limits <- function(
```

Comment:
where will this be called? will it be used to generate an inputdata file in fullREMIND.R?

Reply:
At a later point yes.  Currently it is still under development.